tools/test/config_test/config_test.py

```diff
@@ -39,13 +39,17 @@ def test_tree(full_name, name):
     if "test_data" in sys.modules:
        del sys.modules["test_data"]
     import test_data
+    # If the test defines custom targets, they must exist in a file called
+    # "targets.json" in the test's directory.
+    if os.path.isfile(os.path.join(full_name, "targets.json")):
+        set_targets_json_location(os.path.join(full_name, "targets.json"))
+    else: # uset the regular set of targets
+        set_targets_json_location(Target._Target__targets_json_location_default)
```

Comment:
I'm pretty sure you can leave off the argument to `set_targets_json_location` and have the same behavior. This would prevent you from relying on a 'hiden' class variable.